src/topology/connected.lean

```diff
@@ -132,6 +133,73 @@ begin
     Hs.is_preconnected Ht.is_preconnected
 end
 
+/-- The filtered sUnion of a set S of preconnected subsets is preconnected.
+(A set of sets is filtered if if the inclusion relation is total on it.)
+-/
+theorem is_preconnected.filtered_sUnion_of {S : set (set α)}
+  (K : ∀ s t, s ∈ S → t ∈ S → s ⊆ t ∨ t ⊆ s)
+  (H : ∀ s ∈ S, is_preconnected s ) : is_preconnected ⋃₀ S :=
+begin
+  rw is_preconnected,
+  rintros u v hu hv Huv ⟨a, ⟨s, hsS, has⟩, hau⟩ ⟨b, ⟨t, htS, hbt⟩, hbv⟩,
+  revert a b u v,
+  /- WLOG, we will only prove the case where s ⊆ t -/
+  wlog Hst: s ⊆ t := K s t hsS htS using [s t, t s] tactic.skip,
+  intros a b u v hu hv Huv hau has hbv hbt,
+  have hat : a ∈ t := Hst has,
+  have Hnuv : (t ∩ (u ∩ v)).nonempty,
+    from H t htS u v hu hv ((subset_sUnion_of_mem htS).trans Huv) ⟨a, hat, hau⟩ ⟨b, hbt, hbv⟩,
+  have Ktuv : ( t ∩ (u ∩ v)) ⊆ sUnion S ∩ (u ∩ v),
+    from inter_subset_inter_left _ (subset_sUnion_of_mem htS),
+  exact Hnuv.mono Ktuv,
+
+  /- We now prove t ⊆ s by symmetry -/
+  intros s t a b u v hu hv Huv hau has hbv hbt,
+  rw union_comm at Huv,
+  rw inter_comm u v,
+  exact this t s b a v u hv hu Huv hbv hbt hau has,
+end
```

Comment:
```suggestion
/-- The directed sUnion of a set S of preconnected subsets is preconnected. -/
theorem is_preconnected.sUnion_directed {S : set (set α)}
  (K : directed_on (⊆) S)
  (H : ∀ s ∈ S, is_preconnected s) : is_preconnected (⋃₀ S) :=
begin
  rintros u v hu hv Huv ⟨a, ⟨s, hsS, has⟩, hau⟩ ⟨b, ⟨t, htS, hbt⟩, hbv⟩,
  obtain ⟨r, hrS, hsr, htr⟩ : ∃ r ∈ S, s ⊆ r ∧ t ⊆ r := K s hsS t htS,
  have Hnuv : (r ∩ (u ∩ v)).nonempty,
  from H _ hrS u v hu hv ((subset_sUnion_of_mem hrS).trans Huv) ⟨a, hsr has, hau⟩ ⟨b, htr hbt, hbv⟩,
  have Kruv : r ∩ (u ∩ v) ⊆ ⋃₀ S ∩ (u ∩ v),
  from inter_subset_inter_left _ (subset_sUnion_of_mem hrS),
  exact Hnuv.mono Kruv
end
```
This statement can be generalized by only requiring `S` to be directed, which makes the proof also significantly simpler.